warehouse/query-core/src/main/java/datawave/query/tables/ScannerSession.java

```diff
@@ -403,7 +410,7 @@ protected void findTop() throws Exception {
                 flush();
                 return;
             }
-            stop();
+            stopAsync();
```

Comment:
Not sure exactly what is causing this, but looking over the recent pipeline runs for this PR in Gitlab-CI, I noticed there's a sporadic  failure occurring with the test below, likely due to race condition introduced by the new `stopAsync()` behavior.
```
156768 [ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.666 s <<< FAILURE! - in datawave.webservice.query.runner.QueryExecutorBeanTest
156769 [ERROR] testCloseActuallyCloses(datawave.webservice.query.runner.QueryExecutorBeanTest)  Time elapsed: 0.173 s  <<< FAILURE!
java.lang.AssertionError: 
createQuery thread died before reaching initialize: java.lang.AssertionError: 
  Unexpected method call Persister.remove(datawave.webservice.query.QueryImpl@5d9ef7be[queryLogicName=EventQueryLogic,queryName=<null>,expirationDate=Tue Oct 08 18:23:15 GMT 2019,uuid=962018b3-5610-4add-ab31-4fb007a55da3,pagesize=10,pageTimeout=0,maxResultsOverride=NA,query=FOO == BAR,queryAuthorizations=PRIVATE,PUBLIC,userDN=CN=Guy Some Other soguy, OU=MY_SUBDIVISION, OU=MY_DIVISION, O=ORG, C=US,owner=<null>,parameters=[],dnList=<null>,columnVisibility=<null>,beginDate=Mon Oct 07 18:23:15 GMT 2019,endDate=Mon Oct 07 18:23:15 GMT 2019]):
    Persister.create("cn=guy some other soguy, ou=my_subdivision, ou=my_division, o=org, c=us", [cn=guy some other soguy, ou=my_subdivision, ou=my_division, o=org, c=us], PRIVATE|PUBLIC, "EventQueryLogic", datawave.webservice.query.QueryParametersImpl@d9a26113, {audit.userDn=[cn=guy some other soguy, ou=my_subdivision, ou=my_division, o=org, c=us], audit.columnVisibility=[PRIVATE|PUBLIC], audit.logicClass=[EventQueryLogic]}): expected: 1, actual: 1
    Persister.findById(<any>): expected: at least 0, actual: 0
	at datawave.webservice.query.runner.QueryExecutorBeanTest.testCloseActuallyCloses(QueryExecutorBeanTest.java:733)
```
Even though the build passes after one or more retries, it probably needs some investigation to figure out whether there's an actual issue with the code here, or if test above just needs to accommodate the new behavior. FWIW, now that you've rebased against the latest master, the deploy phase and it's query tests seem to pass consistently on the first try. Although, I'd be curious to see if Wildfly logs are seeing any new stacktraces.

Reply:
I have yet to see that test Fail locally for me but I do see ```MaxExpansionRegexQueryTest``` fail on occasion.

Reply:
> I have yet to see that test Fail locally for me but I do see `MaxExpansionRegexQueryTest` fail on occasion.

To follow up on this, I've recently observed the exact same failure with `testCloseActuallyCloses` on existing branches, so that issue is obviously unrelated to changes here. In light of that, can't see any reason not to approve this

Reply:
Sounds good.  I rebased on current master and merged the conflict